uniffi_bindgen/src/interface/ffi.rs

```diff
@@ -35,6 +35,11 @@ pub enum FFIType {
     /// If you've got one of these, you must call the appropriate rust function to free it.
     /// This is currently only used for error messages, and may go away in future.
     RustCString,
+    /// A `*const T` pointer belonging to a rust-owned `Arc<T>`.
+    /// If you've got one of these, you must call the appropriate rust function to free it.
+    /// The appropriate function will depend on the specific type `T` (which is maybe an
+    /// argument that we should include T in the tyep here, but that's for future...)
+    RustArcPtr,
```

Comment:
Not thrilled about the name, but an explicit "pointer" type of some sort seems worthwhile here.